tests/test_oc4ids_tranforms.py

```diff
@@ -0,0 +1,1332 @@
+                {"id": "1", "name": "a", "roles": ["administrativeEntity"]},
+                {"id": "2", "name": "b", "roles": ["administrativeEntity"]},
+            ],
+        },
+    ]
+
+    output = transforms._run_transforms(
+        copy.deepcopy(releases),
+        "1",
+        transforms=[transforms.contracting_process_setup, transforms.administrative_entity],
+    )
+
+    assert output["parties"] == releases[0]["parties"]
+
+    # tender is not created as there are multiple adminastrative entities
+    assert "tender" not in output["contractingProcesses"][0]["summary"]
+
+
+def test_contract_status_pre_award():
+
+    releases = [
+        {"ocid": "ocds-213czf-1", "id": "1", "tag": "planning", "date": "2001-02-03T04:05:06Z", "tender": {"id": 1}},
+        {
+            "ocid": "ocds-213czf-2",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "contracts": [{"status": "pending"}],
+            "awards": [{"status": "pending"}],
+            "tender": {"id": 1},
+        },
+        {
+            "ocid": "ocds-213czf-3",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "contracts": [{"status": "active"}],
+            "awards": [{"status": "pending"}],
+            "tender": {"id": 1},
+        },
+        {
+            "ocid": "ocds-213czf-4",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "awards": [{"status": "pending", "date": "3000-01-01"}],
+            "tender": {"id": 1, "awardPeriod": {"startDate": "3000-01-01"}},
+        },
+        {
+            "ocid": "ocds-213czf-5",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "awards": [{"date": "3000-01-01"}],
+            "tender": {"id": 1, "awardPeriod": {"startDate": "2000-01-01"}},
+        },
+        {
+            "ocid": "ocds-213czf-6",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "awards": [{"date": "2000-01-01"}],
+            "tender": {"id": 1, "awardPeriod": {"startDate": "3000-01-01"}},
+        },
+    ]
+
+    output = transforms._run_transforms(
+        copy.deepcopy(releases), "1", transforms=[transforms.contracting_process_setup, transforms.contract_status],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["status"] == "pre-award"
+    assert output["contractingProcesses"][1]["summary"]["status"] == "pre-award"
+    assert output["contractingProcesses"][2]["summary"]["status"] != "pre-award"
+    assert output["contractingProcesses"][3]["summary"]["status"] == "pre-award"
+
+    # Currently no status at all as fits no status
+    assert output["contractingProcesses"][4]["summary"].get("status") != "pre-award"
+    assert output["contractingProcesses"][5]["summary"].get("status") != "pre-award"
```

Comment:
I think this one should be "active", since the actual `award/date` is in the past, even though the expected `tender/awardPeriod/startDate` is in the future

I've opened an issue to review and clarify the business logic documented in the codelist: https://github.com/open-contracting/infrastructure/issues/209